fortuna/prob_model/posterior/sgmcmc/sgmcmc_posterior.py

```diff
@@ -52,14 +54,19 @@ def sample(
 
     def load_state(self, checkpoint_dir: Path) -> None:
         try:
-            self.restore_checkpoint(pathlib.Path(checkpoint_dir) / "0")
+            state = self.restore_checkpoint(pathlib.Path(checkpoint_dir) / "c")
         except ValueError:
             raise ValueError(
                 f"No checkpoint was found in `checkpoint_dir={checkpoint_dir}`."
             )
-        self.state = PosteriorMultiStateRepository(
+        which_params = decode_encoded_tuple_of_lists_of_strings_to_array(
+            state._encoded_which_params
+        )
+        self.state = SGMCMCPosteriorStateRepository(
```

Comment:
Is this state repository used also to save the checkpoints while MCMC is running? If so, all good.